src/java.base/share/classes/sun/security/jca/ProviderList.java

```diff
@@ -385,6 +403,11 @@ public Service getService(String type, String name) {
                 Provider p = getProvider(pList.get(i).provider);
                 Service s = p.getService(type, name);
                 if (s != null) {
+                    if (!RestrictedSecurity.isServiceAllowed(s)) {
+                        // We're in restricted security mode which does
+                        // not allow this service, return null.
+                        return null;
+                    }
```

Comment:
Like below, I think we want to just ignore services that are not allowed; one from a later provider may be allowed.
```java
                if ((s != null) && RestrictedSecurity.isServiceAllowed(s)) {
                    // We found a service that is allowed in restricted security mode.
                    return s;
                }
```
Same comment for 420... below.